File: src/main/java/demo/interop/Exceptions.java
package demo.interop;

import java.io.IOException;

public class Exceptions {

    public static void throwIOExceptionFromJava() throws IOException {
        throw new IOException();
    }

    public static void main(String[] args) {
        try {
            ExceptionsKt.throwIOExceptionFromKotlin();
        } catch (IOException e) {

        }
    }

}

File: src/main/kotlin/demo/interop/Exceptions.kt
package demo.interop

import java.io.IOException


@Throws(IOException::class)
fun throwIOExceptionFromKotlin() {
    throw IOException()
}

fun main(args: Array<String>) {

    throwIOExceptionFromKotlin()

    try {
        Exceptions.throwIOExceptionFromJava()
    } catch (e: IOException) {

    }

}